src/rounded_rect.rs

```diff
@@ -0,0 +1,302 @@
+//! A rectangle with rounded corners.
+
+use crate::{arc::ArcAppendIter, Arc, PathEl, Point, Rect, Shape, Vec2};
+use std::f64::consts::{FRAC_PI_2, PI};
+
+/// A rectangle with rounded corners.
+#[derive(Clone, Copy, Default, Debug)]
+pub struct RoundedRect {
+    /// Coordinates of the rectangle.
+    pub rect: Rect,
+    /// Radius of all four corners.
+    pub radius: f64,
+}
+
+impl RoundedRect {
+    /// A new rectangle from minimum and maximum coordinates.
+    #[inline]
+    pub fn new(x0: f64, y0: f64, x1: f64, y1: f64, radius: f64) -> RoundedRect {
+        RoundedRect {
+            rect: Rect::new(x0, y0, x1, y1),
+            radius,
+        }
+    }
+
+    /// A new rectangle from two points.
+    ///
+    /// The result will have non-negative width and height.
+    #[inline]
+    pub fn from_points(p0: Point, p1: Point, radius: f64) -> RoundedRect {
+        RoundedRect {
+            rect: Rect::from_points(p0, p1),
+            radius,
+        }
+    }
+
+    /// A new rectangle from origin and size.
+    ///
+    /// The result will have non-negative width and height.
+    #[inline]
+    pub fn from_origin_size(origin: Point, size: Vec2, radius: f64) -> RoundedRect {
```

Comment:
The method in `Rect` that this is copied from is wrong, the `size` argument can have negative components :/